letsencrypt/cli.py

```diff
@@ -304,31 +304,91 @@ def _auth_from_domains(le_client, config, domains, plugins):
 
     return lineage
 
+class ConfiguratorError(TypeError): # pylint: disable=missing-docstring
+    pass
+
+def set_configurator(previously, now):
+    """Setting configurators multiple ways is okay, as long as they all agree"""
+    if now is None:
+        # we're not actually setting anything
+        return previously
+    if previously:
+        if previously != now:
+            msg = "Too many flags setting configurators/installers/authenticators %s -> %s"
+            raise ConfiguratorError, msg % (`previously`, `now`)
+    return now
+
+def diagnose_configurator_problem(cfg_type, requested):
+    """
+    Raise the most helpful error message about a plugin being unavailable
+    @cfg_type -- "installer" or "authenticator"
```

Comment:
nit: Should be Sphinx-style documentation as used through the rest of the project.